app/javascript/hyacinth_v1/components/shared/forms/buttons/ResponsiveSubmitButton.js

```diff
@@ -1,48 +1,45 @@
-import React from 'react';
+import React, { useState } from 'react';
 import { Button } from 'react-bootstrap';
 import { withRouter } from 'react-router-dom';
 
-class ResponsiveSubmitButton extends React.Component {
-  state = {
-    isSaving: false,
-    success: null,
-  }
+const ResponsiveSubmitButton = (props) => {
+  const { formType, saveData, staticContext, ...rest } = props;
+  const [isSaving, setIsSaving] = useState(false);
+  const [success, setSuccess] = useState(null);
 
-  onClick = (e) => {
+  const onClick = (e) => {
     e.preventDefault();
 
-    this.setState({ isSaving: true });
-    const { saveData } = this.props;
-
-    saveData()
-      .then(() => this.setState({ success: true }))
-      .catch(() => this.setState({ success: false }))
-      .then(() => setTimeout(() => this.setState({ isSaving: false, success: null }), 1000));
+    const exec = () => {
+      setIsSaving(true);
+      saveData();
```

Comment:
I could be wrong, but is saveData usually an asynchronous function that returns a promise?  Meaning that the two lines below this one will run immediately after the function fires and not wait until the async process completes?

Reply:
Restored asynch processing and commented on halting state updates to prevent memory leaks.

Reply:
👍